Tests/RevenueCatUITests/CustomerCenter/ManageSubscriptionsViewModelTests.swift

```diff
@@ -71,27 +73,256 @@ class ManageSubscriptionsViewModelTests: TestCase {
         expect(viewModel.subscriptionInformation?.durationTitle) == "month"
         expect(viewModel.subscriptionInformation?.price) == "$2.99"
-        expect(viewModel.subscriptionInformation?.nextRenewalString) == "Apr 12, 2062"
-        expect(viewModel.subscriptionInformation?.productIdentifier) == "com.revenuecat.product"
+
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        expect(viewModel.subscriptionInformation?.nextRenewalString) ==
+            formatter.string(from: ISO8601DateFormatter().date(from: expirationDate)!)
+        expect(viewModel.subscriptionInformation?.productIdentifier) == productId
+    }
+
+    func testShouldShowEarliestExpiration_whenUserHasTwoActiveSubscriptionsOneEntitlement() async {
+        // Arrange
+        let productIdOne = "com.revenuecat.product1"
+        let productIdTwo = "com.revenuecat.product2"
+        let purchaseDate = "2022-04-12T00:03:28Z"
+        let expirationDateFirst = "2062-04-12T00:03:35Z"
+        let expirationDateSecond = "2062-05-12T00:03:35Z"
+        let products = [
+            Fixtures.product(id: productIdOne, title: "yearly", duration: .year, price: 29.99),
+            Fixtures.product(id: productIdTwo, title: "monthly", duration: .month, price: 2.99)
+        ]
+        let customerInfo = Fixtures.customerInfo(
+            subscriptions: [
+                Fixtures.Subscription(
+                    id: productIdOne,
+                    store: "app_store",
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateFirst
+                ),
+                Fixtures.Subscription(
+                    id: productIdTwo,
+                    store: "app_store",
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateSecond
+                )
+            ].shuffled(),
+            entitlements: [
+                Fixtures.Entitlement(
+                    entitlementId: "premium",
+                    productId: productIdOne,
+                    purchaseDate: purchaseDate,
+                    expirationDate: expirationDateFirst
+                )
+            ]
+        )
+
+        let viewModel = ManageSubscriptionsViewModel(screen: ManageSubscriptionsViewModelTests.screen,
+                                                     purchasesProvider: MockManageSubscriptionsPurchases(
+                                                        customerInfo: customerInfo,
+                                                        products: products
+                                                     ))
+
+        // Act
+        await viewModel.loadScreen()
+
+        // Assert
+        expect(viewModel.subscriptionInformation).toNot(beNil())
+        expect(viewModel.screen).toNot(beNil())
+        expect(viewModel.state) == .success
+
+        expect(viewModel.subscriptionInformation?.title) == "yearly"
```

Comment:
If subscriptionInformation is nil, all these 3 will fail but the error message won't be that clean. You can use `XCTUnwrap` to unwrap the variable and have a neat error message if there's no value there, i.e.: 
```swift
let subscriptionInformation = try XCTUnwrap(viewModel.subscriptionInformation)
```
(you'll have to add a `throws` in the method's signature, but in tests that's fine and it'll still give you a clean test failure)

Reply:
Ah thanks, TIL! A few lines up we do have this: 

```swift
expect(viewModel.subscriptionInformation).toNot(beNil())
```

Is `XCTUnwrap` the convention over `expect(*).toNot(beNil())`?

Reply:
Yes, we should use `XCTUnwrap` instead of `expect(*).toNot(beNil())`. We don't have linting errors for that so reviews/remembering is the only way. Those three up there are my fault, if you can update them it would be amazing.

Good thing about `XCTUnwrap` as well is that it removes the need to do `?` in subsequent checks

Reply:
Yea sure, no worries! Good to be consistent 💪 Updated: https://github.com/RevenueCat/purchases-ios/pull/4046/commits/f10ea101ead44f475214f97d9034ebccc38dab5a